main.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+function wait () {
+	echo "should we continue? (y|n)"
+	read -r user_input
+	if [ "$user_input" == 'y' ]
+	then
+		echo "continuing..."
+	else
+		echo -e "\e[1;42m Exiting... \e[0m"
+		exit 0
+	fi
+}
+function should () {
+	echo "should I \"$1\" (y|n|e)"
+	read -r user_input
+	if [ "$user_input" == 'y' ]
+	then
+		$1
+		status
+	elif [ "$user_input" == 'n' ]
+	then
+		echo -e "\e[1;43m Skipping... \e[0m"
+	else
+		echo -e "\e[1;42m Exiting... \e[0m"
+		exit 0
+	fi
+}
+function status () {
+	if [ "$?" = '0' ]
+	then
+		echo -e "\e[1;42m Success \e[0m"
+	else
+		echo -e "\e[1;41m Failure \e[0m"
+	fi
+}
+function check_dir () {
+	if [ -d "$1" ]
```

Comment:
as a good practice, to avoid the problem below, and to make the scripts more readable, I would declare variables to describe what is expected as function params: 
function check_dir () {
  dir = $1
  comment = $2
  ...